### DISCUSSION TITLE
need vscode Extension?

### DISCUSSION THREAD

**Discussion by CoffeeChaton** [mmikeww/AHK-v2-script-converter#303] [Ideas]
idea <https://www.autohotkey.com/boards/viewtopic.php?f=60&t=132720>

I think some people would like to be able to call AHK-v2-script-converter in vscode,
and use vscode-diff.

Is anyone interested in writing a vscode extension?
If not, and if the community maintainers do not object, I plan to write a vscode extension when I have time.

blueprint:

1. built-in a releases default <https://github.com/mmikeww/AHK-v2-script-converter/releases>
2. Provide options , let user using the project path they want. like:

  ```jsonc
  {
    // settings.json
    "AHK-v2-script-converter.path": "D:\\ahk\\AHK-v2-script-converter-main\\v1.0.3\\",
    "AHK-v2-script-converter.useVscodeDiff":true
  }
  ```

**Comment by Banaanae** [comment]
So if I understand correctly, you plan on making a VSCode extension that converts to v2 via the cli tool, and then lets users view the diff using VSCode built in tools?

Sounds good, I can add a `-s` flag to suppress all msgbox (or even send msg contents to StdOut)
The cli tool isn't yet compiled and released with the QuickConverter, but I can probably figure something out (although i'd wait until #292 is merged because I want a diff to be bundled with the exe)

**Reply by CoffeeChaton** [reply]
Specifically, when I open the AHK-V1 file, a small button will appear in the upper right corner of vscode, prompting can switch the file to v2, and use the  path of the file to call cli, call the converter, and call vscode diff.Looking forward to #292😀

**Reply by CoffeeChaton** [reply]
or select a small piece of v1 code to generate v2 code directly in the preview pane(like `Easy CodeSnap
` Code screenshot tool)
